hdate/tekufot.py

```diff
@@ -0,0 +1,256 @@
+    - Cheilat Geshamim start date, which differs between the diaspora and Israel.
+    - Halachic prayer periods based on key Jewish holidays and seasonal changes.
+    - Appropriate prayer phrases depending on the current date, tradition, and language.
+"""
+
+import datetime as dt
+from enum import IntEnum
+
+from hdate import converters as conv
+from hdate.hebrew_date import HebrewDate
+from hdate.location import Location
+from hdate.translator import TranslatorMixin
+from hdate.zmanim import Zmanim
+
+
+class Gevurot(TranslatorMixin, IntEnum):
+    """Enum class for the gevurot."""
+
+    MORID = 0
+    MASHIV = 1
+    NEITHER = 2
+
+
+class Geshamim(TranslatorMixin, IntEnum):
+    """Enum class for the geshamim."""
+
+    BARKHEINU = 0
+    BARECH_ALEINU = 1
+
+
+class Tekufot(TranslatorMixin):  # pylint: disable=too-many-instance-attributes
+    """
+    A class that calculates and manages Jewish seasonal times (Tekufot),
+    periods for prayer insertions, and associated halachic dates such as
+    the start of Cheilat Geshamim (requesting rain)."""
+
+    # pylint: disable=too-many-arguments, R0917
+    def __init__(
+        self,
+        date: dt.date = dt.datetime.today(),
+        location: Location = Location(),
+        tradition: str = "israel",
+        language: str = "english",
+    ):
+        """Initialize the Tekufot object."""
+        super().__init__()
+
+        self.date = date
+        self.gregorian_year = date.year
+        self.hebrew_year = self.gregorian_year + 3760
+        self.location = location
+        self.tradition = tradition or (
+            "israel" if not location.diaspora else "diaspora_sephardi"
+        )
+        self.language = language
+
+        # Convert current date to JDN and Hebrew Date
+        jdn = conv.gdate_to_jdn(date)
+        self.hebrew_date = conv.jdn_to_hdate(jdn)
+        self.hebrew_year_p = self.hebrew_date.year
+        self.gregorian_year_p = self.hebrew_year_p - 3760
+
+        # Tekufot calculations
+        self.get_tekufa()
+
+        # Cheilat Geshamim calculation
+        self.get_cheilat_geshamim()
+
+    def get_tekufa(self) -> None:
+        """
+        Calculates the approximate dates and times of the Tekufot.
+        This is a simplified approximation. Traditional calculations may differ.
+        """
+
+        interval_days = 91
+        interval_hours = 7.5  # 7 hours and 30 minutes
+
+        # Start with Tekufa Nissan:
+        # Historically approximated at the spring equinox. For simplicity, assume:
+        # If the Hebrew year corresponds to a Gregorian year before 2100
+        # we take April 7 as a reference, otherwise April 8
+
+        if self.gregorian_year_p < 2100:
+            date_equinox_april = dt.date(self.gregorian_year_p, 4, 7)
+        else:
+            date_equinox_april = dt.date(self.gregorian_year_p, 4, 8)
+
+        # Hours shift depends on leap year cycles
+        hours_delta_nissan = (self.gregorian_year_p % 4) * 6
```

Comment:
I don't think this is correct, don't forget the exceptions for years 100/400/2000. I suggest you use the isleap function from calendar:
```
from calendar import isleap
hours_delta_nissan = int(isleap(self.date.year)) * 6
```